init.lua

```diff
@@ -44,6 +44,13 @@ local sharedtags = {
 local function salvage(tag)
     -- The screen to move the orphaned tag to.
     local newscreen = capi.screen.primary
+    -- the primary screen may be the one that is being
+    -- removed, so try to find a different screen if possible.
+    for s in capi.screen do
```

Comment:
Maybe instead of looping through all the screens, just pick the first one with
```lua
local newscreen = next(capi.screen)
```

Reply:
The first screen has a chance of being the one being removed, this happened to me.

In my case, screen 1 (laptop screen) was still around for the `salvage` function call, but was being removed in favor of screens 2 and 3 (docked monitor screens).

Also screen 1 was `current_screen`, so it wouldn't fix the logic for the use case that wasn't working for me.